### PR TITLE
New Java File Handling Cheat Sheet

### PR DESCRIPTION
**Pull request by rasikapohankar**
IA Page link : https://duck.co/ia/view/java_file_handling_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/java-file-handling.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        "Data Streams",
+        "Byte Streams",
+        "Line Oriented I/O"        
+    ],
+
+    "sections": {
+        "Byte Streams": [
+            {
+                "key": "FileInputStream in = \nnew FileInputStream(\"xanadu.txt\");\n\n int c; \n\n while ((c = in.read()) != -1) \\{\n\tSystem.out.println(c);\n\\}\n\n in.close();",
+                "val": "Uses byte stream to perform input of 8 bits at a time. Read method returns integer or long."
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/java-file-handling.json]: @rasikapohankar Lets remove periods at end of val fields.

Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/java-file-handling.json]: @jophab: Hi! What purpose would that serve?

**Comment by moollaza** [comment]
CheatSheets weren't really intended for code blocks like this or lengthy descriptions.

Instead of code blocks can we just give examples of the important functions and explain what they do?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=java file handling cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/java_file_handling_cheat_sheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=java file handling cheat sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/java_file_handling_cheat_sheet_mobile.png)

**Comment by rasikapohankar** [comment]
@moollaza I have modified the contents. Can you check?

File: share/goodie/cheat_sheets/json/java-file-handling.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            {
+                "key": "FileOutputStream out = \nnew FileOutputStream(\"outagain.txt\");",
+                "val": "Uses byte stream to perform output of 8 bits."
+            },
+            {
+                "key": "FileOutputStream out = \nnew FileOutputStream(\"outagain.txt\",true);",
+                "val": "This form of the constructor appends data you write instead of overwriting old data. "
+            },
+            {
+                "key": "Methods: read() ; write(int b) or \nwrite(byte\\[\\] b)",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/java-file-handling.json]: I think these should be three entries in a new section. It feels like we're putting a little too much into 1 item.

**Byte Streams:**
`key`: val
`key`: val
**Byte Stream Methods:**
`read()`: _"explanation"_
`write()`: _"explanation"_

File: share/goodie/cheat_sheets/json/java-file-handling.json [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            {
+                "key": "DataInputStream in = new DataInputStream(\nnew BufferedInputStream(\n new FileInputStream(\"input.txt\")));",
+                "val": "Data streams support binary I/O of primitive data type values. DataInputStream can only be used as a wrapper for an existing byte stream."
+            },
+            {
+                "key": "DataOutputStream out = \nnew DataOutputStream(\nnew BufferedOutputStream(\n new FileOutputStream(dataFile)));",
+                "val": "A DataOutputStream can only be created as a wrapper for an existing byte stream object."
+            },
+            {
+                "key": "Methods- readInt(),readDouble(),readUTF() ; writeInt(int b), writeDouble(double d), writeUTF(String s)",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/java-file-handling.json]: Same here and below -- can we break these up into separate key-value items?

We should reduce a single key and value to as little info as possible to make them very simple and easy to read/understand

File: share/goodie/cheat_sheets/json/java-file-handling.json [hunk 4]
```diff
@@ -0,0 +1,95 @@
+{
+    "id": "java_file_handling_cheat_sheet",
+    "name": "Java File Handling",
+
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/java-file-handling.json]: no `meta`? Is there no source used for this info, or website they can go to read more details?

**Comment by moollaza** [comment]
@rasikapohankar For the first version, this LGTM 👍  -- I've left some comments about improvements.

Feel free to share this with others from the Java community to get their feedback!

**Comment by daxtheduck** [comment]
@rasikapohankar this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by rasikapohankar** [comment]
@moollaza Thanks! I had not separated the methods as there are many more classes under Byte Stream that I didn't know and haven't used. But they all have those methods, I checked now. Will separate it. To make the suggested changes I'll open another pull request? Also where can I find the Java community?

**Pull request merged**